src/interpreter/environment.cpp

```diff
@@ -109,25 +113,25 @@ gc_ptr<Boolean> Environment::new_boolean(bool b)
 gc_ptr<String> Environment::new_string(std::string s)
 { return m_gc->new_string(std::move(s)); }
 
-Array* Environment::new_list(ArrayType elements)
+gc_ptr<Array> Environment::new_list(ArrayType elements)
 { return m_gc->new_list_unsafe(std::move(elements)); }
```

Comment:
Shouldn't you have deleted the "unsafe" methods by this point, and change them for the regular ones?

Reply:
oops they shouldn't be used anymore, but i thought it might be useful to keep them around. I just removed them anyways